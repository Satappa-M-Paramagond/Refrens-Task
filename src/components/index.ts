export { FormHeader } from './header';
export { PrimaryButton } from './buttons';
