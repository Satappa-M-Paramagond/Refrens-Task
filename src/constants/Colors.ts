export const Colors = {
    black: '#000000',
    cyan: 'cyan',
    darkCyan: '#004053',
    darkGrey: 'darkgrey',
    grey: 'grey',
    lightCyan: '#87CBCF',
    lighterCyan: '#364C4D',
    lightGrey: 'lightgrey',
    transparent: 'transparent',
    white: '#ffffff',
    whiteSmoke: 'whitesmoke'
};
