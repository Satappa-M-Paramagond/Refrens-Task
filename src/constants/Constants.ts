export const Constants = {
    angerLevelStatus: ['Low', 'Low', 'Low', 'Medium', 'High', 'High']
};

export const TestIDs = {
    bar1: 'bar1',
    bar2: 'bar2',
    bar3: 'bar3',
    bar4: 'bar4',
    bar5: 'bar5',
    primaryButton: 'primaryButton'
};
