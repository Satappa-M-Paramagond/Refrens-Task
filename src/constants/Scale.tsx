import { Dimensions } from 'react-native';
const { width } = Dimensions.get('window');

// Sample device width to calculate scaling
const guidelineBaseWidth = 375;

export const Scale = (size: number) =>
    size + ((width / guidelineBaseWidth) * size - size) * 0.5;
