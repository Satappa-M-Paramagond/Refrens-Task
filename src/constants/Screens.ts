export const Screens = {
    HOME: 'Home',
    PROFILE: 'Profile',
    RESIDENTS: 'Residents'
};
