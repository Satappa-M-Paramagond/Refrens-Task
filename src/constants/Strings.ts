export const Strings = {
    high: 'High',
    low: 'Low',
    medium: 'Medium',
    next: 'Next',
    pickAngerLevel: 'Pick the level your anger & \nfrustation right now',
    rescureSession: 'Rescue session: Anger & Frustation'
};
