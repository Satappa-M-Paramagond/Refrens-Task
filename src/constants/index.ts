export { Colors } from './Colors';
export { Constants, TestIDs } from './Constants';
export { Scale } from './Scale';
export { Strings } from './Strings';
