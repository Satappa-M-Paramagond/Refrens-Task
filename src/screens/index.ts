export { HomeController } from './home';
export { ProfileController } from './profile';
export { ResidentsController } from './residents';
