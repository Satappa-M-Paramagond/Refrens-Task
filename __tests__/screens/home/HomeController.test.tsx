import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { HomeController } from '../../../src/screens';
import { mount, shallow } from 'enzyme';
import { TestIDs } from '../../../src/constants';

describe('HomeController => Snapshots => ', () => {
    test('should render properly', () => {
        const tree = renderer.create(<HomeController />).toJSON();
        expect(tree).toMatchSnapshot();
        expect.assertions(1);
    });
});

describe.skip('HomeController => Enzyme => ', () => {
    test('should update state on click', () => {
        const setActivePhase = jest.fn();
        const wrapper = mount(<HomeController />);
        const handleSet = jest.spyOn(React, 'useState');
        handleSet.mockImplementation(activePhase => [
            activePhase,
            setActivePhase
        ]);

        wrapper
            .findWhere(n => n.prop('testID') === TestIDs.primaryButton)
            .first()
            .props()
            .onPress();

        expect(setActivePhase).toHaveBeenCalled();
        expect.assertions(1);
    });
});
