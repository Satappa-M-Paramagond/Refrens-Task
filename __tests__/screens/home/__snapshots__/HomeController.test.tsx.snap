// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeController => Snapshots =>  should render properly 1`] = `
<View
  style={
    Object {
      "height": "100%",
      "justifyContent": "space-between",
      "padding": 22.5,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": 3,
            "height": 6,
            "opacity": 1,
            "width": 120,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "darkgrey",
            "borderRadius": 3,
            "height": 6,
            "opacity": 1,
            "width": 120,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "darkgrey",
            "borderRadius": 3,
            "height": 6,
            "opacity": 1,
            "width": 120,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "darkgrey",
            "borderRadius": 3,
            "height": 6,
            "opacity": 1,
            "width": 120,
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "darkgrey",
            "borderRadius": 3,
            "height": 6,
            "opacity": 1,
            "width": 120,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 30,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontSize": 19.5,
            "fontWeight": "bold",
            "textTransform": "uppercase",
          }
        }
      >
        Rescue session: Anger & Frustation
      </Text>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ïˆ£
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontSize": 24,
          "fontWeight": "700",
          "lineHeight": 36,
          "marginTop": 45,
          "textAlign": "left",
        }
      }
    >
      Pick the level your anger & 
frustation right now
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": -120,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "darkgrey",
          "borderRadius": 210,
          "borderStyle": "dashed",
          "borderWidth": 6,
          "height": 420,
          "justifyContent": "center",
          "width": 420,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 172.5,
            "height": 345,
            "justifyContent": "center",
            "width": 345,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#364C4D",
              "borderRadius": 165,
              "height": 330,
              "justifyContent": "center",
              "width": 330,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "cyan",
                "borderRadius": 112.5,
                "height": 225,
                "justifyContent": "center",
                "width": 225,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 66,
                  "fontWeight": "bold",
                }
              }
            >
              0
            </Text>
          </View>
        </View>
      </View>
    </View>
    <RNCSlider
      disabled={false}
      enabled={true}
      inverted={false}
      maximumTrackTintColor="#ffffff"
      maximumValue={5}
      minimumTrackTintColor="cyan"
      minimumValue={0}
      onChange={[Function]}
      onRNCSliderSlidingComplete={null}
      onRNCSliderSlidingStart={null}
      onRNCSliderValueChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      step={1}
      style={
        Array [
          Object {
            "height": 40,
          },
          Object {
            "height": 75,
            "marginTop": 75,
            "width": "90%",
          },
        ]
      }
      tapToSeek={false}
      thumbImage={null}
      thumbTintColor="cyan"
      value={0}
    />
  </View>
  <View
    accessibilityLabel="primaryButton"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderRadius": 33,
        "height": 66,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID="primaryButton"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#004053",
            "fontSize": 22.5,
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      Next
    </Text>
  </View>
</View>
`;
